### PR TITLE
[GStreamer] Stop pushing buffers when seeking status changes.

### PR DESCRIPTION
**Pull request by charlie-ht**
After switching to splitting buffers into smaller block sizes in

   https://bugs.webkit.org/show_bug.cgi?id=182829

It was found that during the individual buffer pushes, the seeking status could
change behind our backs from another thread. When this happens, buffers from
incorrect offsets would find their way into appsrc and eventually the demuxer
itself, which would start parsing from a random place and at best give a
confusing error message.

The solution here is break from pushing buffers when the seeking status has
been detected as changed. Flushes will clear out what we've already delivered
into the appsrc, just have to make sure to not continue sending buffers in
there after the flush.

* platform/graphics/gstreamer/WebKitWebSourceGStreamer.cpp:
(CachedResourceStreamingClient::dataReceived):

### CODE REVIEW

**Comment by charlie-ht** [comment]
I've included some minor code cleanups I found while starting at this code.

File: Source/WebCore/platform/graphics/gstreamer/WebKitWebSourceGStreamer.cpp [hunk 1]
```diff
@@ -997,10 +997,14 @@ void CachedResourceStreamingClient::dataReceived(PlatformMediaResource&, const c
         GST_TRACE_OBJECT(src, "Pushing buffer of size %" G_GSIZE_FORMAT " bytes", gst_buffer_get_size(subBuffer));
         GstFlowReturn ret = gst_app_src_push_buffer(priv->appsrc, subBuffer);
 
-       if (UNLIKELY(ret != GST_FLOW_OK && ret != GST_FLOW_EOS && ret != GST_FLOW_FLUSHING)) {
+        if (UNLIKELY(ret != GST_FLOW_OK && ret != GST_FLOW_EOS && ret != GST_FLOW_FLUSHING)) {
+            gst_buffer_unref(subBuffer);
```
Comment (calvaris) [inline Source/WebCore/platform/graphics/gstreamer/WebKitWebSourceGStreamer.cpp]: This line asks for a `GRefPtr<GstBuffer>`  with a `leakRef()` in the `gst_app_src_push_buffer`, doesn't it?

File: Source/WebCore/platform/graphics/gstreamer/WebKitWebSourceGStreamer.cpp [hunk 2]
```diff
@@ -976,9 +976,9 @@ void CachedResourceStreamingClient::dataReceived(PlatformMediaResource&, const c
     // to push buffers that are too large, otherwise incorrect buffering messages can be sent from the
     // pipeline.
     uint64_t bufferSize = gst_buffer_get_size(priv->buffer.get());
```
Comment (calvaris) [inline Source/WebCore/platform/graphics/gstreamer/WebKitWebSourceGStreamer.cpp]: This does not belong to this patch so I guess we can forget, but just a quick heads up: we don't need to call `gst_buffer_get_size` because unless I am missing something in lines 959-962, it is ensured that `bufferSize == length`, sign apart...

File: Source/WebCore/platform/graphics/gstreamer/WebKitWebSourceGStreamer.cpp (hunk 1 above)
Comment (charlie-ht) [reply Source/WebCore/platform/graphics/gstreamer/WebKitWebSourceGStreamer.cpp]: My feelings are the lifetime of this variable is very clear, and wrapping it in a smart pointer would cause more ugliness for little conceptual gain. If your feelings are stronger than mine, I'm happy to change it :)

File: Source/WebCore/platform/graphics/gstreamer/WebKitWebSourceGStreamer.cpp (hunk 2 above)
Comment (charlie-ht) [reply Source/WebCore/platform/graphics/gstreamer/WebKitWebSourceGStreamer.cpp]: That's true, it's probably clearer to call the size API, but can be cleaned up.

**Comment by eocanha** [comment]
The changes look good to me after @calvaris comments have been addressed.

**Comment by charlie-ht** [comment]
I have moved the buffer to being handled in GRefPtr
